openmdao/recorders/test/test_dump.py

```diff
@@ -422,6 +423,38 @@ def test_driver_records_metadata(self):
 
         self.assertMetadataRecorded((expected_params, expected_unknowns, expected_resids))
 
+    def test_driver_records_unknown_types_metadata(self):
+        
+        prob = Problem()
+        root = prob.root = Group()
+
+        # Need an optimization problem to test to make sure 
+        #   the is_desvar, is_con, is_obj metadata is being 
+        #   recorded for the Unknowns
+        root.add('p1', IndepVarComp('x', 50.0), promotes=['*'])
+        root.add('p2', IndepVarComp('y', 50.0), promotes=['*'])
+        root.add('comp', Paraboloid(), promotes=['*'])
```

Comment:
Good point. But we probably need to put in a story to clean up all uses of promotes in the testing code. I am sure there's a bunch. I just copied this from another test.